### DISCUSSION TITLE
Multiple messages for different recipients in one request

### DISCUSSION THREAD

**Discussion by dorkyboi** [ctwillie/expo-server-sdk-php#7] [Q&A] [answered]
Hi, i am wondering how can I accumulate messages and send them in one go?
This would appear to be a basic functionality, since Expo themselves [state at their docs](https://docs.expo.dev/push-notifications/sending-notifications/#http2-api) that amount of requests should be minimal.
As far as I know, your code allows to add only one message per request that may be sent to multiple recipients.

As a use case, let's imagine a script that fetches all new grades for all students.
Now, because there is no way to specify recepients for a group of messages or on a per-message basis, I would have to make as many requests as there are students that had new grades added (because notifications' bodies are unique for each student).
Plus, the amount grows even bigged in case if I had to send a couple of different messages to same student.

Is there any way to add a bunch of messages (potentially with different contents) with predefined recipients?

This could be done relatively easily, by:
- allowing `to` property for `ExpoMessage` instances
- making `Expo->recipients` behave as _default_ onces (applied if `ExpoMessage->to` is empty)
- adding method that would allow adding multiple messages (like `->message()` but for an array of messages)

This would not even be a breaking change, if done correctly.

**Comment by ctwillie** [comment]
Hey @dorkyboi!

Correct. You can only send one message per request with the current implementation. Sending multiple messages in a single request is definitely a feature I would like to be present.

Interested in a PR? If not, I plan on adding this feature as soon as I'm able.

Thanks,

**Reply by dorkyboi** [reply]
I've just released a PR #8. This should cover aforementioned feature and also introduce a couple quality of life stuff.
Hope we can merge it soon, my current project could really use it.

**Comment by dorkyboi** [comment, marked as answer]
Said feature was implemented in PR #8 and is available since v1.3.

**Reply by ctwillie** [reply]
@dorkyboi 

Thanks for the contribution! 🥇
